src/test/java/com/wikia/webdriver/testcases/discussions/PollsTests.java

```diff
@@ -0,0 +1,158 @@
+        postsCreator.addPoll(); //create new poll
+        Assert.assertEquals(poll.getAnswersInputsList().size(), DEFAULT_ANSWERS_NUMBER);
+
+        poll.addTitle(TextGenerator.createUniqueText());
+        poll.addNthAnswer(TextGenerator.createUniqueText(), 0);
+        poll.addNthAnswer(TextGenerator.createUniqueText(), 1);
+        poll.clickAddAnswerButton();
+        poll.addNthAnswer(TextGenerator.createUniqueText(), 2);
+        poll.clickAddAnswerButton();
+        poll.addNthAnswer(TextGenerator.createUniqueText(), 3);
+        Assert.assertEquals(poll.getAnswersInputsList().size(), DEFAULT_ANSWERS_NUMBER + 2);
+
+        poll.deleteNthAnswer(0);
+        Assert.assertEquals(poll.getAnswersInputsList().size(), DEFAULT_ANSWERS_NUMBER + 1);
+
+        postsCreator.clickSubmitButton();
+        page.waitForLoadingSpinner();
+        Assert.assertTrue(new PostsListPage().getPost().firstPostHasPoll());
+    }
+
+    @Test
+    @InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5)
+    @Execute(asUser = User.USER_6, onWikia = MercuryWikis.DISCUSSIONS_2)
+    public void userCanCreatePostWithSimplePollOnMobile() {
+        BasePostsCreator postsCreator = new PostsListPage().open().getPostsCreatorMobile();
+        postsCreator.click().closeGuidelinesMessage().clickAddCategoryButton().selectFirstCategory();
+        postsCreator.addTitleWith(TextGenerator.createUniqueText());
+        Poll poll = postsCreator.addPoll();
+        Assertion.assertFalse(postsCreator.isPostButtonActive());
+
+        poll.addTitle(TextGenerator.createUniqueText());
+        Assertion.assertFalse(postsCreator.isPostButtonActive());
+        poll.addNthAnswer(TextGenerator.createUniqueText(), 0);
+        Assert.assertFalse(postsCreator.isPostButtonActive());
+        poll.addNthAnswer(TextGenerator.createUniqueText(), 1);
+        postsCreator.clickSubmitButton();
+
+        Assert.assertTrue(new PostsListPage().getPost().firstPostHasPoll());
+    }
+
+    @Test(dependsOnMethods = {"userCanCreatePostWithSimplePollOnMobile"})
+    @InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5)
+    @Execute(asUser = User.ANONYMOUS, onWikia = MercuryWikis.DISCUSSIONS_2)
+    public void anonUserCanNotVoteInPollOnMobile() {
+        Poll poll = new PostsListPage().open().getPost().clickNthPostWithPoll(0).getPoll();
+        poll.clickPollTitle();
+        SignInToFollowModalDialog signInModal = new SignInToFollowModalDialog();
+        Assert.assertTrue(signInModal.isVisible());
+
+        signInModal.clickOkButton();
+        poll.clickNthAnswer(0);
+        Assert.assertTrue(signInModal.isVisible());
+    }
+
+    @Test(dependsOnMethods = {"userCanCreatePostWithSimplePollOnMobile"})
+    @InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5)
+    @Execute(asUser = User.USER_3, onWikia = MercuryWikis.DISCUSSIONS_2)
+    public void loggedInUserCanVoteOnceInPollOnMobile() {
```

Comment:
mobile and desktop tests are more of less the same - please re-use the code to avoid duplications

Reply:
Tbh I made this consciously. Since WebElements for Mobile & Desktop are the same (because of the framework under testing) methods are also the same for both views. IMO there's nothing wrong wit two tests being almost the same and just using other emulator. I see two solutions - splitting suites per Mobile/Desktop or encapsulating some methods in helper on the tests side. First is painful because of amount of class needed to be created (e.g. Discussion project), second is in my opinion just adding one more abstraction layer to shut up the static code analyser. ;) 

Perfectly In the future we could have sth like (which IMO is the good solution here):
`@Test
@InBrowser(emulator = {Emulator.DESKTOP_BREAKPOINT_BIG, emulator = Emulator.GOOGLE_NEXUS_5})
public void fooBar() {
}`